src/hatti/map/utils.cljs

```diff
@@ -506,6 +506,87 @@
                                   :line-cap "round"}}
     :else {:layer-type "circle" :style :point}))
 
+(defn to-fixed
+  [n number-of-decimal-places]
+  (when n
+    (-> n
+        (.toFixed number-of-decimal-places)
+        js/parseFloat)))
+
+(defn percent
+  [n total & {:keys [number-of-decimal-places]}]
+  (let [percentage (when (and (pos? n) (>= total n))
+                     (-> n
+                         (/ total)
+                         (* 100)
+                         float))]
+    (if number-of-decimal-places
+      (to-fixed percentage number-of-decimal-places)
+      percentage)))
+
+(defn generate-hexgrid
+  "Generates hexbins with point count aggregation given rendered
+  layer-id or geojson"
+  [map layer-id & [geojson]]
+  (let [get-rendered-features #(.queryRenderedFeatures
+                                map (clj->js {:layers [layer-id]}))
+        rendered-features (or geojson
+                              (clj->js
+                               {:type "FeatureCollection"
+                                :features (get-rendered-features)}))
+        ;; Get bounding box for rendered features
+        bbox (.bbox js/turf (clj->js rendered-features))
+        cellWidth 60
```

Comment:
where does this number come from? should either be an arg or in a descriptive def